cpp/src/parquet/encoding.cc

```diff
@@ -2546,11 +2364,13 @@ int ByteStreamSplitDecoder<DType>::DecodeArrow(
     } else {
       builder->UnsafeAppendNull();
     }
-    bit_reader.Next();
-  }
+  };
+
+  VisitNullBitmapInline(valid_bits, valid_bits_offset, num_values, null_count,
+                        std::move(decode_value));
 
   num_values_ -= values_decoded;
-  len_ -= sizeof(num_streams) * values_decoded;
+  len_ -= sizeof(T) * values_decoded;
```

Comment:
```suggestion
  len_ -= kNumStreams * values_decoded;
```

Reply:
`sizeof(T)` is logically more exact, though it's the same value (the length in bytes is decreased by the number of values decoded times the value size).